Library/Homebrew/utils/curl.rb

```diff
@@ -367,6 +377,88 @@ def curl_supports_tls13?
     def http_status_ok?(status)
       (100..299).cover?(status.to_i)
     end
+
+    # Separates the output text from `curl` into an array of response heads and
+    # the final response body.
+    # @param output [String] The output text from `curl` containing
+    #   response head(s), body, or both.
+    # @return [Hash] A hash containing an array of the response heads and the
+    #   body output, if found.
+    sig { params(output: String).returns(T::Hash[Symbol, T.untyped]) }
+    def parse_curl_output(output)
+      heads = []
+      return { heads: heads, body: "" } unless output.is_a?(String)
+
+      max_iterations = 5
+      iterations = 0
+      output = output.lstrip
+      while output.match?(%r{\AHTTP/[\d.]+ \d+}) && output.include?(HTTP_HEAD_BODY_SEPARATOR)
+        iterations += 1
+        raise "Too many redirects (max = #{max_iterations})" if iterations > max_iterations
+
+        head_text, _, output = output.partition(HTTP_HEAD_BODY_SEPARATOR)
+        output = output.lstrip
+        next if head_text.blank?
+
+        head_text.chomp!
+        head = parse_curl_head(head_text)
+        heads << head if head.present?
+      end
+
+      { heads: heads, body: output }
+    end
+
+    # Returns the URL from the last location header found in cURL output
+    # response heads.
+    # @param heads [Array<Hash>] An array of hashes containing response status
+    #   information and headers from `parse_curl_head`.
+    # @param url [String, nil] The URL to use as a base for making the
+    #  `location` URL absolute.
+    # @param absolutize [true, false] Whether to make the location URL absolute.
+    # @return [String, nil] The URL from last-occurring `location` header, if
+    #   any, in the responses.
+    sig {
+      params(
+        heads:      T::Array[T::Hash[Symbol, T.untyped]],
+        url:        T.nilable(String),
+        absolutize: T::Boolean,
+      ).returns(T.nilable(String))
+    }
+    def curl_response_last_location(heads, url: nil, absolutize: false)
+      heads.reverse_each do |head|
+        next if !head[:headers] || !(location = head[:headers]["location"])
+
+        return (absolutize && url.is_a?(String)) ? URI.join(url, location).to_s : location
```

Comment:
```suggestion
        return (absolutize && url.is_a?(String)) ? URI.join(url, location).to_s : location
```

A bit too complex for a ternary here, let's make this an `if`/`else`. Similarly: when is this/is it not a `String`? Feels like it'd be nicer to not have to sniff types here.

Reply:
Using an `if`/`else` with the `return` here leads to some issues. If we don't use parentheses around the `if`/`else`:

```ruby
return if absolutize && url.is_a?(String)
  URI.join(url, location).to_s
else
  location
end
```

...we receive a `SyntaxError` (`else without rescue is useless`).

If we wrap it in parentheses, `brew style` complains but running `brew style --fix` leads to some gnarly styling:

```ruby
return (if absolutize && url.is_a?(String)
          URI.join(url, location).to_s
        else
          location
end)
```

`brew style` stubbornly wants the `end)` in line with the `return` and will complain if we move the `end)` in line with the `if`/`else`.

With all that in mind, what do you think about this approach?:

```ruby
absolute_url = URI.join(url, location).to_s if absolutize && url.present?
return absolute_url || location
```

I think it's comprehensible while still being relatively terse (and arguably more visually appealing than the `if`/`else` setup).

---

I must have overlooked this `url.is_a?(String)` condition when updating the others. I'll update this to `url.present?` and simply rely on the Sorbet type signature to enforce the argument type.

Reply:
> With all that in mind, what do you think about this approach?:

Looks good!